AWSAuthSDK/Sources/AWSAuthCore/AWSAppleSignIn/AWSAppleSignInButton.m

```diff
@@ -0,0 +1,81 @@
+//
+// Copyright 2010-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+#import "AWSAppleSignInButton.h"
+#import "AWSAppleSignInProvider.h"
+
+@interface AWSAppleSignInButton()
+
+@property (nonatomic, strong) id<AWSSignInProvider> signInProvider;
+
+@property (nonatomic, strong) UIControl *appleButton;
+@property (nonatomic, strong) UIStackView *appleStackView;
+
+@end
+
+@implementation AWSAppleSignInButton
+
+@synthesize delegate;
+@synthesize buttonStyle;
```

Comment:
I understand they're declared in the protocol--that's actually where I copied my suggested code from :)

My point is that rather than simply declaring a bare `@synthesize` of these symbols, I'd prefer to be explicit about the properties we're using, local to the file. I've also seen projects that use a `#pragma mark` to annotate synthesis of protocol properties, but I prefer being more explicit.

The downside of course is that we could drift out of sync if the protocol definition changes, so if you want to keep the `@synthesize`, please annotate with a comment to note the types of these properties.

Reply:
Added annotation